frontstage/views/surveys.py

```diff
@@ -1,181 +1,155 @@
+                           collection_instrument_size=case_data['collection_instrument_size'],
+                           survey_info=case_data['survey'],
+                           collection_exercise_info=case_data['collection_exercise'],
+                           business_info=case_data['business_party'],
+                           referer_header=referer_header)
 
 
 @surveys_bp.route('/download_survey', methods=['GET'])
 @jwt_authorization(request)
 def download_survey(session):
-    party_id = session.get('party_id', 'no-party-id')
-    collection_instrument_id = request.args.get('cid')
-    case_id = request.args.get('case_id')
-    loggerb = logger.bind(party_id=party_id, collection_instrument_id=collection_instrument_id, case_id=case_id)
-    case = get_case(case_id, loggerb)
-    case_party_id = case.get('partyId')
+    party_id = session['party_id']
+    case_id = request.args['case_id']
+    logger.info('Downloading collection instrument', case_id=case_id, party_id=party_id)
 
-    valid = access_surveys_permissions(party_id, case_party_id, loggerb)
-    if not valid:
-        return render_template("errors/error.html", _theme='default', data={"error": {"type": "failed"}})
+    params = {
+        "case_id": case_id,
+        "party_id": party_id
+    }
+    response = api_call('GET', app.config['DOWNLOAD_CI'], parameters=params)
 
-    collection_instrument, status_code, headers = download_collection_instrument(collection_instrument_id, logger)
-    post_download_collection_instrument_case_event(case_id, party_id, collection_instrument_id, status_code)
+    if response.status_code != 200:
+        logger.error('Failed to download collection instrument', party_id=party_id, case_id=case_id)
+        raise ApiError(response)
 
-    if status_code == 200:
-        logger.info('Successfully downloaded collection instrument', collection_instrument_id=collection_instrument_id)
-        return collection_instrument, status_code, headers
-    else:
-        logger.error('Failed to download collection instrument', status_code=status_code)
-        return render_template('surveys/surveys-download-failure.html',
-                               _theme='default',
-                               error_info=request.args.get('error_info', None))
+    logger.info('Sucessfully downloaded collection instrument', case_id=case_id, party_id=party_id)
+    return response.content, response.status_code, response.headers.items()
 
 
 @surveys_bp.route('/upload_survey', methods=['POST'])
 @jwt_authorization(request)
 def upload_survey(session):
-    """Logged in page for users only."""
-    party_id = session.get('party_id', 'no-party-id')
-    case_id = request.args.get('case_id', None)
-    logger.info('Attempting to upload survey', party_id=party_id, case_id=case_id)
-
-    if request.content_length > app.config['MAX_UPLOAD_LENGTH']:
-        error_info = "size"
```

Comment:
Did we agree on keeping this check here as well as/instead of in frontsage-api?

Reply:
I'll add this check back into frontstage now, We probably need a more robust way to be validating the files though

Reply:
Yeah for sure